packages/config/src/normalize.js

```diff
@@ -28,7 +28,7 @@ const DEFAULT_CONFIG = {
 }
 
 const normalizePlugin = function ({ inputs = {}, ...plugin }) {
-  return { ...plugin, inputs }
+  return removeFalsy({ ...plugin, inputs })
```

Comment:
This removes `undefined`, `null` and empty strings (after trimming) values.
See https://github.com/netlify/build/blob/b8204b6f89607a5ff42811016282af64e381ebb5/packages/config/src/utils/remove_falsy.js#L6